solidserver/soliderver.go

```diff
@@ -75,18 +102,62 @@ func (s *SOLIDserver) GetVersion(version string) error {
 		log.Printf("[DEBUG] Cert Subjects After Append = %d\n", len(rootCAs.Subjects()))
 	}
 
+	t := httpRequestTimings[method]
+	log.Printf("[DEBUG] timings for method '%s' : {%v}\n", method, t)
+
+	apiclient.Timeout(time.Duration(t.sTimeout) * time.Second)
+
+	retryCount := 0
+
+KeepTrying:
+	for retryCount < t.maxTry {
+
+		log.Printf("[DEBUG] request retryCount=%d\n", retryCount)
+
+		httpFunc, ok := httpRequestMethods[method]
+		if !ok {
+			return nil, "", fmt.Errorf("SOLIDServer - Error initiating API call, unsupported HTTP request '%s'\n", method)
+		}
+		// Random Delay for write operation to distribute the load
+		time.Sleep(time.Duration(rand.Intn(t.msSweep)) * time.Millisecond)
+		requestUrl = fmt.Sprintf("%s/%s?%s", s.BaseUrl, service, parameters)
+		resp, body, errs = httpFunc(apiclient, requestUrl).
+			TLSClientConfig(&tls.Config{InsecureSkipVerify: !s.SSLVerify, RootCAs: rootCAs}).
+			Set("X-IPM-Username", base64.StdEncoding.EncodeToString([]byte(s.Username))).
+			Set("X-IPM-Password", base64.StdEncoding.EncodeToString([]byte(s.Password))).
+			End()
+
+		if errs == nil {
+			return resp, body, nil
+		}
+
+		log.Printf("[DEBUG] '%s' API request '%s' failed with errors...\n", method, requestUrl)
+		for i, err := range errs {
+			log.Printf("[DEBUG] errs[%d] / (%s) = '%v'\n", i, reflect.TypeOf(err), err)
+			// https://stackoverflow.com/questions/23494950/specifically-check-for-timeout-error/23497404
+			if err, ok := err.(net.Error); ok && err.Timeout() {
```

Comment:
Just trying to understand this piece of code.

If i comment "log.Printf("[DEBUG] errs[%d] / (%s) = '%v'\n", i, reflect.TypeOf(err), err)", I get a compilation error "invalid type assertion: err.(net.Error) (non-interface type int on left)" related to the line that follows.

I'm a bit surprise, since this call is not supposed to modify the variable 'err'.

What's is your purpose doing the following test: "err.(net.Error)" ?

I seem to get it working using the following:

```
if err, ok := reflect.TypeOf(err).(net.Error); ok && err.Timeout() {
```

Makes sense ?

Reply:
@alexissavin beyond my Golang knowledge, if itworks I'm happy with it :-) 

My intent was just to check that the error is a connection timeout, hence the type assertion to `net.Error` and the test of the return value of Timeout() and all I did was to blindly paste the line from the StackOverflow link.

I've also been startled by this error when I first wrote the patch and didn't relalize this `reflect.TypeOf(err)` has been hidden the problem since.

I would definitely love to understand this quirk though, I'll try to dig the interweb on this matter.

Reply:
@aderuelle my bad, I mis interpreted the code and got confused with the behavior of the golang range function :)

Merged and working fine (it does seems to improve the performance of the provider, but I don't have measurement of this yet).

Many thanks for your contribution.

Kind regards